flink-runtime/src/test/java/org/apache/flink/runtime/rpc/TestingRpcService.java

```diff
@@ -51,7 +52,11 @@
  * verify(testGateway, timeout(1000)).theTestMethod(any(UUID.class), anyString());
  * }</pre>
  */
-public class TestingRpcService extends AkkaRpcService {
+public class TestingRpcService implements RpcService {
+
+    // load RpcSystem only once to save initialization costs
+    // this is safe because it is state-less
+    private static final RpcSystem RPC_SYSTEM_SINGLETON = RpcSystem.load();
```

Comment:
we could think about adding a JVM shutdown hook because this does leak a 50mb temp file, but it's a temp file so ... 🤷

Reply:
Agree, it's a temp file, so probably not worth too much effort. Also, the one implementation of `RpcSystem` deletes it on calling `close()`, so as long as tests make sure to call close, it should even do that as is.